### ISSUE TITLE
Add step type to histograms

### ISSUE THREAD

**Issue by aloctavodia** [arviz-devs/arviz-plots#96]
Stepped histograms, instead of bars histograms, make it slightly easier to overlap and compare multiple distributions.

**Comment by OriolAbril** [comment]
I agree we should have this option. Not really sure about defaults though, personally I find easier to read overlaid bar histograms with transparency. Easier which is not easy though, both break multiple gestalt principles of visual perception and require a lot of concentration to see the continuity of each distribution.

**Comment by The-Broken-Keyboard** [comment]
Heyy

If this issue is still open then I would like to work on this.

Please assign me this issue 

Thank you

**Comment by aloctavodia** [comment]
assigned

**Comment by The-Broken-Keyboard** [comment]
heyy @aloctavodia 

I was implementing the step histogram for various backends and I have reached the conclusion that step histogram functionality is only provided by matplotlib and neither bokeh nor plotly provide any functionality like step histograms. 

Do you want me to implement for matplotlib only and leave it for other two backends?

**Comment by aloctavodia** [comment]
It should be implemented for all or none of the backends. There is no direct option but you can do something like this https://docs.bokeh.org/en/latest/docs/user_guide/basic/lines.html#step-lines and then add two lines.

For plotly not sure. What about this https://stackoverflow.com/questions/64453989/plotly-how-to-plot-histogram-in-root-style-showing-only-the-contours-of-the-his

**Comment by The-Broken-Keyboard** [comment]
Okay, these workarounds look feasible.

Will do these for bokeh and plotly

**Comment by The-Broken-Keyboard** [comment]
@aloctavodia I think we can close this now, since the functionality has been implemented now
